node-repository/src/main/java/com/yahoo/vespa/hosted/provision/maintenance/NodeFailer.java

```diff
@@ -326,41 +329,48 @@ private boolean throttle(Node node) {
         if (throttlePolicy == ThrottlePolicy.disabled) return false;
         Instant startOfThrottleWindow = clock.instant().minus(throttlePolicy.throttleWindow);
         List<Node> nodes = nodeRepository().getNodes();
-        long recentlyFailedNodes = nodes.stream()
-                .filter(n -> n.history().hasEventAfter(History.Event.Type.failed, startOfThrottleWindow))
-                .count();
-        int allowedFailedNodes = (int) Math.max(nodes.size() * throttlePolicy.fractionAllowedToFail,
-                                                throttlePolicy.minimumAllowedToFail);
-
-        boolean throttle = allowedFailedNodes < recentlyFailedNodes ||
-                (allowedFailedNodes == recentlyFailedNodes && !node.type().isDockerHost());
-        if (throttle) {
-            log.info(String.format("Want to fail node %s, but throttling is in effect: %s", node.hostname(),
-                                   throttlePolicy.toHumanReadableString()));
-        }
-        metric.set(throttlingActiveMetric, throttle ? 1 : 0, null);
-        return throttle;
+        NodeList recentlyFailedNodes = nodes.stream()
+                                            .filter(n -> n.history().hasEventAfter(History.Event.Type.failed, startOfThrottleWindow))
+                                            .collect(collectingAndThen(Collectors.toList(), NodeList::new));
+
+        // Allow failing nodes within policy
+        if (recentlyFailedNodes.size() < throttlePolicy.allowedToFailOf(nodes.size())) return false;
+
+        // Always allow failing parents up to minimum limit
+        if (!node.parentHostname().isPresent() &&
```

Comment:
My comment got somehow lost due to the force-push:

Consider using `node.type().isDockerHost()` for parent check, it also handles infrastructure hosts and wont give false positives like BM tenant nodes.

Reply:
That's precisely why I used the `parentHostname().isPresent()` check, BM tenant nodes should not be throttled in this case.

Reply:
Ah, maybe the comments should refer to this as physical hosts rather than parents then?

Reply:
Sure.